### PR TITLE
support `tidb_enable_plan_replayer` variable

### PR DESCRIPTION
**Pull request by Yisaer**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-plan-replayer.md [hunk 1]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
 ```
 
 加载并还原所需现场后，即可在该现场诊断和改进执行计划。
+
+## `PLAN REPLAYER CAPTURE` 抓取目标计划
+
+在用户定位 TiDB 执行计划的部分场景中，目标 SQL 与目标计划的可能在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取，此时我们可以使用 `PLAN REPLAYER CAPTURE` 来帮助我们定向抓取目标 SQL 与计划的优化器信息情况。
```
Comment (ran-huang) [inline sql-plan-replayer.md:179]:
```suggestion
在用户定位 TiDB 执行计划的部分场景中，目标 SQL 与目标计划可能仅在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取。此时你可以使用 `PLAN REPLAYER CAPTURE` 来帮助你定向抓取目标 SQL 与目标计划的优化器信息情况。
```

File: sql-plan-replayer.md [hunk 2]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+
+`PLAN REPLAYER CAPTURE` 主要功能如下：
+
+- 在 TiDB 集群内部提前注册目标 SQL 与 PLAN 的 Digest，并开始匹配目标查询。
+- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
+
+### 开启 `PLAN REPLAYER CAPTURE` 功能
+
+`PLAN REPLAYER CAPTURE` 功能通过[`tidb_enable_plan_replayer_capture`](/system-variables.md#tidb_enable_plan_replayer_capture) 控制。
```
Comment (ran-huang) [inline sql-plan-replayer.md:188]:
```suggestion
`PLAN REPLAYER CAPTURE` 功能通过 [`tidb_enable_plan_replayer_capture`](/system-variables.md#tidb_enable_plan_replayer_capture) 控制。
```

File: sql-plan-replayer.md [hunk 3]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+
+### 使用 `PLAN REPLAYER CAPTURE` 功能
+
+我们可以通过以下方式向 TiDB 集群注册目标 SQL 和 PLAN 的 Digest:
```
Comment (ran-huang) [inline sql-plan-replayer.md:192]:
```suggestion
你可以通过以下方式向 TiDB 集群注册目标 SQL 和计划的 Digest:
```

File: sql-plan-replayer.md [hunk 4]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' 'plan_digest';
+```
+
+当我们的目标 SQL 有多种 PLAN 对应且都想抓取时，我们可以通过以下 SQL 一键注册:
```
Comment (ran-huang) [inline sql-plan-replayer.md:198]:
```suggestion
当你的目标 SQL 有多种执行计划对应，且你想抓取所有执行计划时，你可以通过以下 SQL 一键注册:
```

File: sql-plan-replayer.md [hunk 5]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' '*';
+```
+
+### 获取 `PLAN REPLAYER CAPTURE` 结果
+
+当 `PLAN REPLAYER CAPTURE` 成功抓取到结果后，可以通过以下 SQL 查看用于下载的文件标识:
```
Comment (ran-huang) [inline sql-plan-replayer.md:206]: How can a user know whether `PLAN REPLAYER CAPTURE` has successfully captured the result? Is there a way to tell?

File: sql-plan-replayer.md [hunk 6]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+| 086e3fbd2732f7671c17f299d4320689deeeb87ba031240e1e598a0ca14f808c | 042de2a6652a6d20afc629ff90b8507b7587a1c7e1eb122c3e0b808b1d80cc02 |            | replayer_Utah4nkz2sIEzkks7tIRog==_1668746293523179156.zip | 2022-11-18 12:38:13 | NULL        | 172.16.4.4:4022 |
+| b5b38322b7be560edb04f33f15b15a885e7c6209a22b56b0804622e397199b54 | 1770efeb3f91936e095f0344b629562bf1b204f6e46439b7d8f842319297c3b5 |            | replayer_Z2mUXNHDjU_WBmGdWQqifw==_1668746293560115314.zip | 2022-11-18 12:38:13 | NULL        | 172.16.4.4:4022 |
+| 96d00c0b3f08795fe94e2d712fa1078ab7809faf4e81d198f276c0dede818cf9 | 8892f74ac2a42c2c6b6152352bc491b5c07c73ac3ed66487b2c990909bae83e8 |            | replayer_RZcRHJB7BaCccxFfOIAhWg==_1668746293578282450.zip | 2022-11-18 12:38:13 | NULL        | 172.16.4.4:4022 |
++------------------------------------------------------------------+------------------------------------------------------------------+------------+-----------------------------------------------------------+---------------------+-------------+-----------------+
+3 rows in set (0.00 sec)
+```
+
+下载 `PLAN REPLAYER CAPTURE` 的文件方法与 `PLAN REPLAYER` 相同。
```
Comment (ran-huang) [inline sql-plan-replayer.md:220]:
```suggestion
下载 `PLAN REPLAYER CAPTURE` 的文件方法与 `PLAN REPLAYER` 相同，请参考 [`PLAN REPLAYER` 导出示例](#plan-replayer-导出示例)。
```

File: system-variables.md [hunk 7]
```diff
@@ -1636,6 +1636,13 @@ Query OK, 0 rows affected (0.09 sec)
 - 这个变量用来控制是否开启窗口函数的支持。默认值 1 代表开启窗口函数的功能。
 - 由于窗口函数会使用一些保留关键字，可能导致原先可以正常执行的 SQL 语句在升级 TiDB 后无法被解析语法，此时可以将 `tidb_enable_window_function` 设置为 `OFF`。
 
+### `tidb_enable_plan_replayer_capture`
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`FALSE`
+- 这个变量用来控制是否开启 plan replayer capture 功能的开启。默认值 0 代表关闭 plan replayer capture 的功能。
```
Comment (ran-huang) [inline system-variables.md:1644]:
```suggestion
- 这个变量用来控制是否开启 [`PLAN REPLAYER CAPTURE` 功能](/sql-plan-replayer.md)。默认值 `FALSE` 代表关闭 `PLAN REPLAYER CAPTURE` 功能。
```

File: sql-plan-replayer.md [hunk 8]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+`PLAN REPLAYER CAPTURE` 主要功能如下：
+
+- 在 TiDB 集群内部提前注册目标 SQL 与 PLAN 的 Digest，并开始匹配目标查询。
+- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
+
+### 开启 `PLAN REPLAYER CAPTURE` 功能
+
+`PLAN REPLAYER CAPTURE` 功能通过[`tidb_enable_plan_replayer_capture`](/system-variables.md#tidb_enable_plan_replayer_capture) 控制。
+
+### 使用 `PLAN REPLAYER CAPTURE` 功能
```
Comment (songrijie) [inline sql-plan-replayer.md:190]: 如何查看目前正在等待被capture的sql？

File: sql-plan-replayer.md [hunk 9]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+我们可以通过以下方式向 TiDB 集群注册目标 SQL 和 PLAN 的 Digest:
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' 'plan_digest';
+```
+
+当我们的目标 SQL 有多种 PLAN 对应且都想抓取时，我们可以通过以下 SQL 一键注册:
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' '*';
```
Comment (songrijie) [inline sql-plan-replayer.md:201]: SQL 持续被执行的时候，每个 pair 只会被 capture 一次？

File: sql-plan-replayer.md [hunk 10]
```diff
@@ -173,3 +173,48 @@ mysql> show stats_meta;
...
+```
+
+### 获取 `PLAN REPLAYER CAPTURE` 结果
```
Comment (songrijie) [inline sql-plan-replayer.md:204]: 文件的保存策略是什么？ 默认保留多久？ 如何清理？

File: sql-plan-replayer.md (hunk 9 above)
Comment (Yisaer) [reply sql-plan-replayer.md:201]: 对的

File: sql-plan-replayer.md (hunk 10 above)
Comment (Yisaer) [reply sql-plan-replayer.md]: updated.

File: sql-plan-replayer.md (hunk 8 above)
Comment (Yisaer) [reply sql-plan-replayer.md:190]: updated.

File: sql-plan-replayer.md (hunk 5 above)
Comment (Yisaer) [reply sql-plan-replayer.md:223]: 没发现结果那就是没抓到

File: sql-plan-replayer.md [hunk 11]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+当你的目标 SQL 有多种执行计划对应，且你想抓取所有执行计划时，你可以通过以下 SQL 一键注册:
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' '*';
+```
+
+### 查看 `PLAN REPLAYER CAPTURE` 抓取任务
+
+我们可以通过以下方式查看集群中目前正在工作的 `PLAN REPLAYER CAPTURE` 的抓取任务:
```
Comment (ran-huang) [inline sql-plan-replayer.md:206]:
```suggestion
你可以通过以下方式查看集群中目前正在工作的 `PLAN REPLAYER CAPTURE` 的抓取任务:
```

File: sql-plan-replayer.md [hunk 12]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
++-------------+--------------+---------------------+
+1 row in set (0.01 sec)
+```
+
+### 查看 `PLAN REPLAYER CAPTURE` 抓取结果
+
+当 `PLAN REPLAYER CAPTURE` 成功抓取到结果后，可以通过以下 SQL 查看用于下载的文件标识:
+
+```sql
+mysql> select * from mysql.plan_replayer_status;
```
Comment (ran-huang) [inline sql-plan-replayer.md:226]:
```suggestion
mysql> SELECT * FROM mysql.plan_replayer_status;
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 13]
```diff
@@ -1505,14 +1505,10 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 ### `tidb_enable_plan_replayer_capture`
 
-> 警告：
->
-> 当前版本中该变量控制的功能尚未完全生效，请保留默认值。
-
 - 作用域：SESSION | GLOBAL
 - 是否持久化到集群：是
```
Comment (TomShawn) [inline system-variables.md:1509]:
```suggestion
- 是否持久化到集群：是
- 类型：布尔型
```

File: sql-plan-replayer.md [hunk 14]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
 ```
 
 加载并还原所需现场后，即可在该现场诊断和改进执行计划。
+
+## `PLAN REPLAYER CAPTURE` 抓取目标计划
+
+在用户定位 TiDB 执行计划的部分场景中，目标 SQL 与目标计划可能仅在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取。此时你可以使用 `PLAN REPLAYER CAPTURE` 来帮助你定向抓取目标 SQL 与目标计划的优化器信息情况。
```
Comment (TomShawn) [inline sql-plan-replayer.md:179]:
```suggestion
在用户定位 TiDB 执行计划的部分场景中，目标 SQL 语句与目标计划可能仅在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取。此时你可以使用 `PLAN REPLAYER CAPTURE` 来帮助定向抓取目标 SQL 语句与目标计划的优化器信息。
```

File: sql-plan-replayer.md [hunk 15]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+`PLAN REPLAYER CAPTURE` 主要功能如下：
+
+- 在 TiDB 集群内部提前注册目标 SQL 与 PLAN 的 Digest，并开始匹配目标查询。
```
Comment (TomShawn) [inline sql-plan-replayer.md:183]:
```suggestion
- 在 TiDB 集群内部提前注册目标 SQL 语句与执行计划的 Digest，并开始匹配目标查询。
```

File: sql-plan-replayer.md [hunk 16]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
+
+### 开启 `PLAN REPLAYER CAPTURE` 功能
```
Comment (TomShawn) [inline sql-plan-replayer.md:186]:
```suggestion
### 开启 `PLAN REPLAYER CAPTURE`
```

File: sql-plan-replayer.md [hunk 17]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+`PLAN REPLAYER CAPTURE` 功能通过 [`tidb_enable_plan_replayer_capture`](/system-variables.md#tidb_enable_plan_replayer_capture) 控制。
```
Comment (TomShawn) [inline sql-plan-replayer.md:188]:
```suggestion
`PLAN REPLAYER CAPTURE` 功能通过系统变量 [`tidb_enable_plan_replayer_capture`](/system-variables.md#tidb_enable_plan_replayer_capture) 控制。要开启 `PLAN REPLAYER CAPTURE`，将变量值设为 `ON`。
```

File: sql-plan-replayer.md [hunk 18]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+### 使用 `PLAN REPLAYER CAPTURE` 功能
+
+你可以通过以下方式向 TiDB 集群注册目标 SQL 和计划的 Digest:
```
Comment (TomShawn) [inline sql-plan-replayer.md:192]:
```suggestion
你可以通过以下方式向 TiDB 集群注册目标 SQL 语句和计划的 Digest:
```

File: sql-plan-replayer.md [hunk 19]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+```sql
+PLAN REPLAYER CAPTURE 'sql_digest' 'plan_digest';
+```
+
+当你的目标 SQL 有多种执行计划对应，且你想抓取所有执行计划时，你可以通过以下 SQL 一键注册:
```
Comment (TomShawn) [inline sql-plan-replayer.md:198]:
```suggestion
当你的目标 SQL 语句对应多种执行计划，且你想抓取所有执行计划时，你可以通过以下 SQL 语句一键注册:
```

File: sql-plan-replayer.md [hunk 20]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+
+### 查看 `PLAN REPLAYER CAPTURE` 抓取任务
+
+我们可以通过以下方式查看集群中目前正在工作的 `PLAN REPLAYER CAPTURE` 的抓取任务:
+
+```sql
+mysql> plan replayer capture 'example_sql' 'example_plan';
+Query OK, 1 row affected (0.01 sec)
+
+mysql> select * from mysql.plan_replayer_task;
```
Comment (TomShawn) [inline sql-plan-replayer.md:212]:
```suggestion
mysql> PLAN PLAYER CAPTURE 'example_sql' 'example_plan';
Query OK, 1 row affected (0.01 sec)

mysql> SELECT * FROM mysql.plan_replayer_task;
```

File: sql-plan-replayer.md [hunk 21]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
+| sql_digest  | plan_digest  | update_time         |
++-------------+--------------+---------------------+
+| example_sql | example_plan | 2023-01-28 11:58:22 |
++-------------+--------------+---------------------+
+1 row in set (0.01 sec)
+```
+
+### 查看 `PLAN REPLAYER CAPTURE` 抓取结果
+
+当 `PLAN REPLAYER CAPTURE` 成功抓取到结果后，可以通过以下 SQL 查看用于下载的文件标识:
```
Comment (TomShawn) [inline sql-plan-replayer.md:223]:
```suggestion
当 `PLAN REPLAYER CAPTURE` 成功抓取到结果后，可以通过以下 SQL 语句查看用于下载的文件标识:
```

File: sql-plan-replayer.md [hunk 22]
```diff
@@ -173,3 +173,69 @@ mysql> show stats_meta;
...
 加载并还原所需现场后，即可在该现场诊断和改进执行计划。
+
+## `PLAN REPLAYER CAPTURE` 抓取目标计划
+
+在用户定位 TiDB 执行计划的部分场景中，目标 SQL 与目标计划可能仅在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取。此时你可以使用 `PLAN REPLAYER CAPTURE` 来帮助你定向抓取目标 SQL 与目标计划的优化器信息情况。
+
+`PLAN REPLAYER CAPTURE` 主要功能如下：
+
+- 在 TiDB 集群内部提前注册目标 SQL 与 PLAN 的 Digest，并开始匹配目标查询。
+- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
```
Comment (songrijie) [inline sql-plan-replayer.md:184]:
```suggestion
- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
- 针对匹配到的每组 SQL 和 执行计划，信息只抓取一次。
- 通过系统表显示正在进行的匹配任务，以及生成的文件。
- 定时清理历史文件。
```

File: system-variables.md [hunk 23]
```diff
@@ -1505,14 +1505,10 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
-- 类型：布尔型
 - 默认值：`OFF`
+- 这个变量用来控制是否开启 [`PLAN REPLAYER CAPTURE` 功能](/sql-plan-replayer.md)。默认值 `FALSE` 代表关闭 `PLAN REPLAYER CAPTURE` 功能。
```
Comment (ran-huang) [inline system-variables.md:1511]:
```suggestion
- 这个变量用来控制是否开启 [`PLAN REPLAYER CAPTURE` 功能](/sql-plan-replayer.md#使用-plan-replayer-capture-抓取目标计划)。默认值 `FALSE` 代表关闭 `PLAN REPLAYER CAPTURE` 功能。
```

File: system-variables.md [hunk 24]
```diff
@@ -1505,14 +1505,10 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 ### `tidb_enable_plan_replayer_capture`
 
-> 警告：
->
-> 当前版本中该变量控制的功能尚未完全生效，请保留默认值。
-
 - 作用域：SESSION | GLOBAL
```
Comment (songrijie) [inline system-variables.md:1508]: 这个变量是不是应该只有 GLOBAL 的作用域，代表 capture 的任务是否会被触发？

Comment (Yisaer) [reply system-variables.md:1508]: `GLOBAL` 代表集群所有会话都会尝试抓取对应的 capture task 的 Plan replayer 结果，而 `Session` 则代表了只有当前会话会尝试抓取，两者代表了 capture 不同的作用域。

Comment (songrijie) [reply system-variables.md:1508]: LGTM

File: system-variables.md [hunk 25]
```diff
@@ -1513,14 +1513,11 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 
 ### `tidb_enable_plan_replayer_capture`
```
Comment (ran-huang) [inline system-variables.md:1514]: @songrijie @Yisaer Can users change this system variable and use the related feature (`PLAN REPLAYER CAPTURE`) on **TiDB Cloud**? If yes, which tier?
If not, we need to mark this as not applicable to Cloud.

File: sql-plan-replayer.md [hunk 26]
```diff
@@ -173,3 +173,72 @@ mysql> show stats_meta;
...
+
+## 使用 `PLAN REPLAYER CAPTURE` 抓取目标计划
+
+在用户定位 TiDB 执行计划的部分场景中，目标 SQL 语句与目标计划可能仅在查询中偶尔出现，无法使用 `PLAN REPLAYER` 直接抓取。此时你可以使用 `PLAN REPLAYER CAPTURE` 来帮助定向抓取目标 SQL 语句与目标计划的优化器信息。
+
+`PLAN REPLAYER CAPTURE` 主要功能如下：
+
+- 在 TiDB 集群内部提前注册目标 SQL 语句与执行计划的 Digest，并开始匹配目标查询。
+- 当目标查询匹配成功时，直接抓取其优化器相关信息，导出为 ZIP 格式的文件用于保存。
+- 针对匹配到的每组 SQL 和执行计划，信息只抓取一次。
```
Comment (TomShawn) [inline sql-plan-replayer.md:185]:
```suggestion
- 在一次抓取操作中获取匹配到的每组 SQL 和执行计划。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

File: system-variables.md (hunk 25 above)
Comment (Yisaer) [reply system-variables.md:1514]: they can use this feature in Cloud

**Pull request merged**